flink-runtime/src/main/java/org/apache/flink/runtime/operators/ReduceCombineDriver.java

```diff
@@ -42,34 +44,38 @@
  * Combine operator for Reduce functions, standalone (not chained).
  * Sorts and groups and reduces data, but never spills the sort. May produce multiple
  * partially aggregated groups.
- * 
+ *
  * @param <T> The data type consumed and produced by the combiner.
  */
 public class ReduceCombineDriver<T> implements Driver<ReduceFunction<T>, T> {
-	
+
 	private static final Logger LOG = LoggerFactory.getLogger(ReduceCombineDriver.class);
 
 	/** Fix length records with a length below this threshold will be in-place sorted, if possible. */
 	private static final int THRESHOLD_FOR_IN_PLACE_SORTING = 32;
-	
-	
+
+
 	private TaskContext<ReduceFunction<T>, T> taskContext;
 
 	private TypeSerializer<T> serializer;
 
 	private TypeComparator<T> comparator;
-	
+
 	private ReduceFunction<T> reducer;
-	
+
 	private Collector<T> output;
-	
+
+	private DriverStrategy strategy;
+
 	private InMemorySorter<T> sorter;
-	
+
 	private QuickSort sortAlgo = new QuickSort();
 
+	private ReduceHashTable<T> table;
+
 	private List<MemorySegment> memory;
 
-	private boolean running;
+	private volatile boolean canceled;
```

Comment:
Oops, it's `volatile` in the other drivers as well, sorry! So then it can be just volatile here as well, right?